### PR TITLE
[Merged by Bors] - feat: port RingTheory.WittVector.Basic

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

- [x] depends on: #4687

This needs some tactic work that I'm unlikely to get around to

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#4687
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#4687~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by adomani** [comment]
I can't work more on this file for the next several hours: feel free to take over, if you want!

**Comment by jcommelin** [comment]
I'm happy to merge this, but maybe someone with a bit more meta skills wants to take a look.

cc @semorrison

**Comment by adomani** [comment]
I have some local changes that I wanted to push: `git log` shows this on my computer
```bash
commit 1835bd16c12076f68cba7df6e8d0f2fa2658d32d (HEAD -> port/RingTheory.WittVector.Basic, origin/port/RingTheory.WittVector.Basic)
Merge: d7f7bb130 1ae835c77
Author: adomani <adomani@gmail.com>
Date:   Wed Jun 7 18:36:37 2023 +0200

    Merge remote-tracking branch 'origin/master' into port/RingTheory.WittVector.Basic

commit d7f7bb13072dea12ed832132c7eefe2e7fa8b084
Author: adomani <adomani@gmail.com>
Date:   Wed Jun 7 18:35:38 2023 +0200

    Format, to update git?

commit 2e0ff0ab5a459f0852d8b89510e2296d2ead9e71
Author: adomani <adomani@gmail.com>
Date:   Wed Jun 7 18:27:32 2023 +0200

    Remove unnecessary `<;>`

commit e83df9e0230b919dd67c4451e3d402b45222cf04
Author: adomani <adomani@gmail.com>
Date:   Wed Jun 7 18:26:30 2023 +0200

    Fix capitalization in comment

commit 25cc216f57f24688098660e3a5badbd79e63911e
Author: adomani <adomani@gmail.com>
Date:   Wed Jun 7 18:24:54 2023 +0200

    Extend `map_fun_tac` to solve `one`.

commit 9d3751bf9eb57636d1dd9d5ee90bf66bb2f78255
Author: adomani <adomani@gmail.com>
Date:   Wed Jun 7 18:14:42 2023 +0200

    Update porting note: `map_fun_tac` works on `zero`

commit 4717606ff677d999d2274cbd9943e213be58d85d
Author: adomani <adomani@gmail.com>
Date:   Wed Jun 7 18:10:37 2023 +0200

    One more rename in a comment

commit d9bb7a3a3359934f80cca128d91ec19fe7c0a29d
Author: adomani <adomani@gmail.com>
Date:   Wed Jun 7 18:07:34 2023 +0200

    auto: naming

```
However, the PR on GitHub is stuck at 4717606ff677d999d2274cbd9943e213be58d85d...

File: Mathlib/RingTheory/WittVector/Basic.lean [hunk 1]
```diff
@@ -62,235 +62,216 @@ variable {α : Type _} {β : Type _}
...
+  simp only [mapFun, mk, comp_apply, zero_coeff, map_zero,
+    -- porting note: the lemmas on the next line do not have the `simp` tag in mathlib4
+    add_coeff, sub_coeff, mul_coeff, neg_coeff, nsmul_coeff, zsmul_coeff, pow_coeff,
+    peval, map_aeval, algebraMap_int_eq, coe_eval₂Hom]  <;>
+  try { cases n <;> simp <;> done }  <;>  -- porting note: this line solves `one`
+  apply eval₂Hom_congr (RingHom.ext_int _ _) _ rfl  <;>
+  ext ⟨i, k⟩ <;>
+    fin_cases i <;> rfl ))
+
+--  porting note: mathport left a long line, probably due to the presence of meta code, from here
```
Comment (jcommelin) [inline Mathlib/RingTheory/WittVector/Basic.lean:107]:
```suggestion
```
yeah those lines should just be removed, and we don't even leave porting notes about them (-;

File: Mathlib/RingTheory/WittVector/Basic.lean [hunk 2]
```diff
@@ -62,235 +62,216 @@ variable {α : Type _} {β : Type _}
...
+    apply_fun aeval x at this
+    simpa only [aeval_bind₁, aeval_X, ghostFun, aeval_wittPolynomial]
 
 @[local instance]
 private def comm_ring_aux₁ : CommRing (𝕎 (MvPolynomial R ℚ)) :=
-  letI : CommRing (MvPolynomial R ℚ) := MvPolynomial.commRing
-  (ghost_equiv' p (MvPolynomial R ℚ)).Injective.CommRing ghost_fun ghost_fun_zero ghost_fun_one
-    ghost_fun_add ghost_fun_mul ghost_fun_neg ghost_fun_sub ghost_fun_nsmul ghost_fun_zsmul
-    ghost_fun_pow ghost_fun_nat_cast ghost_fun_int_cast
+  -- Porting note: needed? letI : CommRing (MvPolynomial R ℚ) := MvPolynomial.commRing
```
Comment (jcommelin) [inline Mathlib/RingTheory/WittVector/Basic.lean:253]:
```suggestion
  -- Porting note: no longer needed?
  -- letI : CommRing (MvPolynomial R ℚ) := MvPolynomial.commRing
```

**Comment by jcommelin** [comment]
bors d=adomani

**Review by jcommelin — COMMENTED** [review]
bors r+

**Pull request closed without merging**